tests/ops/op_math/test_prod.py

```diff
@@ -1485,20 +1485,6 @@ def circuit(weights):
         true_grad = -qnp.sqrt(2) * qnp.cos(weights[0] / 2) * qnp.sin(weights[0] / 2)
         assert qnp.allclose(grad, true_grad)
 
-    def test_non_hermitian_obs_not_supported(self):
-        """Test that non-hermitian ops in a measurement process will raise a warning."""
-        wires = [0, 1]
-        dev = qml.device("default.qubit", wires=wires)
-        prod_op = Prod(qml.RX(1.23, wires=0), qml.Identity(wires=1))
-
-        @qml.qnode(dev)
-        def my_circ():
-            qml.PauliX(0)
-            return qml.expval(prod_op)
-
-        with pytest.raises(NotImplementedError):
-            my_circ()
```

Comment:
This still raises a `NotImplementedError` 🤔 

```
wires = [0, 1]
dev = qml.device("default.qubit")

print(qml.RX(1.23, wires=0).is_hermitian)

prod_op = qml.ops.op_math.prod(qml.RX(1.23, wires=0), qml.Identity(wires=1))

@qml.qnode(dev)
def my_circ():
    qml.PauliX(0)
    return qml.expval(prod_op)

my_circ()
```

Reply:
```
File [~/Documents/pennylane/pennylane/devices/qubit/measure.py:217](https://file+.vscode-resource.vscode-cdn.net/Users/isaac/Documents/pennylane_pr_QA/~/Documents/pennylane/pennylane/devices/qubit/measure.py:217), in get_measurement_function(measurementprocess, state)
    [214](https://file+.vscode-resource.vscode-cdn.net/Users/isaac/Documents/pennylane_pr_QA/~/Documents/pennylane/pennylane/devices/qubit/measure.py:214)     if measurementprocess.obs is None or measurementprocess.obs.has_diagonalizing_gates:
    [215](https://file+.vscode-resource.vscode-cdn.net/Users/isaac/Documents/pennylane_pr_QA/~/Documents/pennylane/pennylane/devices/qubit/measure.py:215)         return state_diagonalizing_gates
--> [217](https://file+.vscode-resource.vscode-cdn.net/Users/isaac/Documents/pennylane_pr_QA/~/Documents/pennylane/pennylane/devices/qubit/measure.py:217) raise NotImplementedError

NotImplementedError:
```

Reply:
Yes, and it should! Good catch! The description of the test is a bit misleading, so I changed it a bit, but leave the test as this behavior should be continued to be tested.